Assets/Scripts/GameGrid.cs

```diff
@@ -191,71 +193,95 @@ public int FindMatches()
+                // If next tag is null or different, check if counter is > 3 and add match coordinates to matchList
+                int matchCounter = 1;
+                for (int row = 0; row < _grid.GetLength(0); row++)
                 {
-                    if (_grid.GetLength(0) - row < 3)
-                        continue;
-
-                    if (_grid[row, col] == null || _grid[row + 1, col] == null || _grid[row + 2, col] == null)
-                        continue;
+                    if (row + 1 == _grid.GetLength(0) || !_grid[row, col].CompareTag(_grid[row + 1, col].tag))
+                    {
+                        if (matchCounter >= 3)
+                        {
+                            // I did the math, trust me
+                            var rangeStart = row - (matchCounter - 1);
+                            var rangeEnd = row;
 
-                    if (_grid[row, col].CompareTag(_grid[row + 1, col].tag) &&
-                        _grid[row, col].CompareTag(_grid[row + 2, col].tag))
-                        return PerformMatch(row, col, true);
+                            for (int range = rangeStart; range <= rangeEnd; range++)
+                            {
+                                // If already in list, do not add to avoid null reference when removing
+                                var coords = Tuple.Create(range, col);
+                                if (!matchList.Contains(coords))
+                                    matchList.Add(coords);
+                            }
+                        }
+                        matchCounter = 1;
+                    }
+                    else
+                    matchCounter++;
+                    
                 }
             }
-
-            return 0;
+            return PerformMatch(matchList);
         }
 
-        public int PerformMatch(int row, int col, bool IsRowMatch)
+        public int PerformMatch(List<Tuple<int, int>> MatchesList)
         {
             var towerTier = TierTracker.CurrentTier[TierTracker.TierTypes.Tower];
             var treeTier = TierTracker.CurrentTier[TierTracker.TierTypes.Tree];
 
-            var iconType = _grid[row, col].tag;
+            var points = 0;
 
-            // Delete all objects in the found matches
-            RemoveGameObject(row, col);
-
-            if (IsRowMatch)
-            {
-                RemoveGameObject(row + 1, col);
-                RemoveGameObject(row + 2, col);
-            }
-            else
+            foreach (var coord in MatchesList)
             {
-                RemoveGameObject(row, col + 1);
-                RemoveGameObject(row, col + 2);
+                string iconType = _grid[coord.Item1, coord.Item2].tag;
+
+                if (towerTier > 1 && iconType == "Corn")
+                    points += 1 * treeTier + towerTier;
+                else
+                    points += 1 * treeTier;
```

Comment:
Realized points are now per GameObject, not per match. Now they'll be a lot higher.